From 2103f35601127bfb1447da3f149384a81f45bbbe Mon Sep 17 00:00:00 2001
From: Xiang Wang <xiang.a.wang@intel.com>
Date: Mon, 25 Jan 2016 09:11:24 +0800
Subject: [PATCH 6/7] i2c: designware: fix the loop issue in i2c_dw_init

If use "timeout--", when real timeout happens, timeout
equals -1, can't be captured by "timeout == 0"

Change-Id: I5644bfb4e0a7d3bc6b0b81aa8d2f3d09c1be4df8
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-14705
Signed-off-by: Xiang Wang <xiang.a.wang@intel.com>
Reviewed-on: https://android.intel.com:443/461702
---
 drivers/i2c/busses/i2c-designware-core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-designware-core.c b/drivers/i2c/busses/i2c-designware-core.c
index 9c5e9f7..c6b3054 100644
--- a/drivers/i2c/busses/i2c-designware-core.c
+++ b/drivers/i2c/busses/i2c-designware-core.c
@@ -322,7 +322,7 @@ int i2c_dw_init(struct dw_i2c_dev *dev)
 			udelay(10);
 		else
 			usleep_range(10, 100);
-	} while (timeout--);
+	} while (--timeout);
 
 	if (unlikely(timeout == 0))
 		dev_err(dev->dev, "controller time out\n");
-- 
1.9.1

